File: Arrays & Hashing/5. Top K Frequent Elements/solution.scala
object Solution {
  def topKFrequent(nums: Array[Int], k: Int): Array[Int] = {
    nums
      .groupBy(x => x)
      .values
      .toArray
      .sortWith{ case (a,b) => a.length.compareTo(b.length) > 0 }
      .map(_.head)
      .take(k)
  }
}

File: Arrays & Hashing/5. Top K Frequent Elements/solution.exs
defmodule Solution do
  @spec top_k_frequent(nums :: [integer], k :: integer) :: [integer]
  def top_k_frequent(nums, k) do
    nums
    |> Enum.frequencies()
    |> Enum.sort_by(&elem(&1, 1), :desc)
    |> Enum.take(k)
    |> Enum.map(&elem(&1, 0))
  end
end
